### ISSUE TITLE
Directus Integration

### ISSUE THREAD

**Issue by gbtux** [directus/directus#2251]
Hi Team !
Installing a fresh Directus Suite for a Proof Of Concept.
Play with It, create Collections and use some fields and all is beautiful ;)
But....
For my use case, i can't push Admin Interface to end users....
I need to "integrate" Directus Forms in my company portal.

Use Case : 
- An admin create a new Collection and choose fields to create an artifact
- On our company portal, end users create entries based on collections

So, how i can do that  ? Don't want to recreate all the form entry interface (Vue input & co) :/

Just want to integrate this panel (capture) in my portal. Want HTML and not JSON ;)
Thanks !

![capture du 2018-11-22 13-38-47](https://user-images.githubusercontent.com/2924704/48903537-17e3b100-ee5c-11e8-823c-e38f390fd2e6.png)

**Comment by gbtux** [comment]
Quick note:  my portal is powered by Vue JS (2) ;)

**Comment by rijkvanzanten** [comment]
Hey @gbtux! This is not available in the Directus suite by default. There's no way to magically get the correct HTML and CSS of just the edit forms. That being said, you should be able to recreate the edit form in your own app reusing the Vue components that make up the edit form.

All interfaces are separate JS files that you can load into the client side to register them as vue components. Then you can use the information that's being returned in the `/fields` endpoint to render these custom (async) components using a `<component>` component. 

I created a loadExtension function that will asynchronously pull in the Vue component for the given URL which you can use in the `Vue.component()` method:

https://github.com/directus/app/blob/14002ab33852589256d74607b436c83cc74ac358/src/helpers/load-extension.js#L6-L43

Vue.component usage:

https://github.com/directus/app/blob/14002ab33852589256d74607b436c83cc74ac358/src/components/extension/interface/input/input.vue#L162-L166

Each interface uses these props:

https://github.com/directus/app/blob/14002ab33852589256d74607b436c83cc74ac358/src/components/extension/interface/input/input.vue#L2-L21

TL;DR:

* Fetch the fields and interfaces used from `/fields/:collection`
* For each interface, register the interface as a Vue component using `Vue.register()`
* Render the edit form using a loop and `<component :is="interfaceName" />`
* Post the form content to `/items/:id` (how you would regularly update an item)
